### ISSUE TITLE
RRS: Implement RRS functionality

### ISSUE THREAD

**Issue by philpem** [philpem/hylink#3]
It would be useful to have support for the Radio Registration Service in Hylink.

This requires setting up a radio with RRS, finding some dispatch software which implements the protocol and black-boxing it.

As a minimum the format of the RRS request/reply packets needs to be figured out.

**Comment by smarek** [comment]
protocol is described here https://github.com/smarek/Hytera_Homebrew_Bridge/blob/master/kaitai/radio_registration_service.ksy
and parser based on kaitai is generated here https://github.com/smarek/Hytera_Homebrew_Bridge/blob/master/kaitai/radio_registration_service.py

**Comment by philpem** [comment]
Wow, thank you @smarek - and thanks for your other comments too. This is really useful information! And the Kaitai format is really easy to read too, I figured it out without reading any of the documentation.

Did you reverse-engineer the protocol yourself, or did Hytera provide you with the ADK?

**Comment by smarek** [comment]
I've found ADK documentation lying on the baidu shared drives, anyway, if you're up for some fun, I'd really appreciated test data, parser works on binary packets, which I have started putting here: https://github.com/smarek/Hytera_Homebrew_Bridge/tree/master/tests/data
And I need a lot more, to verify that the parser works as expected and under all conditions.
Would you be willing to contribute ?

**Comment by philpem** [comment]
Absolutely. If you can let me know what you need, I can fire up an RD985 and some radios and get some Pcap dumps for you.

Please feel free to drop me an email about that, philpem@gmail.com.
